cmd/cluster/core/create.go

```diff
@@ -922,3 +926,73 @@ func parseTolerationString(str string) (*corev1.Toleration, error) {
 
 	return &toleration, nil
 }
+
+func ValidateMultiArch(ctx context.Context, multiArch bool, opts *CreateOptions) (bool, error) {
+	validMultiArchImage := false
+
+	// Check if the release image is multi-arch
+	if len(opts.ReleaseImage) > 0 && len(opts.PullSecretFile) > 0 {
+		pullSecret, err := os.ReadFile(opts.PullSecretFile)
+		if err != nil {
+			return multiArch, fmt.Errorf("failed to read pull secret file: %w", err)
+		}
+
+		validMultiArchImage, err = registryclient.IsMultiArchManifestList(ctx, opts.ReleaseImage, pullSecret)
+		if err != nil {
+			return multiArch, err
+		}
+	}
+
+	// Check if a release stream was provided instead and its multi-arch
+	if len(opts.ReleaseStream) > 0 && strings.Contains(opts.ReleaseStream, "multi") {
+		validMultiArchImage = true
+	}
+
+	// The release image is not multi-arch so check the mgmt & nodepool cpu arches match
+	if !validMultiArchImage {
+		mgmtClusterCPUArch, err := hyperutil.GetMgmtClusterCPUArch(ctx)
```

Comment:
GetMgmtClusterCPUArch need to be unit tested

Reply:
This method requires a valid mgmt cluster. Is there a way to unit test something like that?

Reply:
As far as I can see it's a client making a request and parsing the response. So you can mock the client response and so document the expectation at the same time